design/open/commit-metadata/airflow.md

```diff
@@ -0,0 +1,180 @@
+# Metadata for Airflow integration
+
+## About
+
+This is a design to integrate metadata from Airflow into lakeFS.  It
+consists of metadata integration and UI for using it.
+
+## Metadata integration
+
+Metadata that Airflow operators and/or hooks may place on commits to
+associate them with the generating Airflow.
+
+### Stability
+
+Commit _metadata_, like diamonds, are forever: It is not generally possible to
+modify commit metadata.  Even if we were willing to rewrite commits and
+could do so safely, it would still break commit digests as stable
+identifiers.
+
+### Forwards compatibility
+
+Obviously we cannot guarantee any future metadata compatibility.  However
+_where current implementation cost is unaffected_, this design is _forwards
+compatible_:
+
+* Airflow metadata added today should remain usable.
+* **If** we ever add UI support for metadata from another (non-Airflow)
+  system, it should work with metadata added today.
+
+### Source of truth
+
+A [wise woman once wrote](https://twitter.com/mipsytipsy/status/998084191488126976):
+
+> You either have one source of truth, or multiple sources of lies.
+
+There is a natural tension and thereby a continuoum between these two
+extremes:
+
+* **Normalized data.**  lakeFS commits hold only enough to identify an
+  Airflow run - its URL.  Airflow holds all other run metadata, and is
+  required to make use of the URL.
+* **Denormalized data.**  lakeFS commits hold all the information that
+  lakeFS and its users need about an Airflow run.  All run metadata is
+  copied to lakeFS commit metadata.
+
+We always need the normalized data pointer, it is the only real identifier
+of the run.[^1] The lakeFS UI can offer the best integration when lakeFS is
+the source of truth.  Initially we shall allow users to select what metadata
+to copy.  The default will be _all_.  For velocity we will avoid adding any
+metadata that is in any way difficult to produce in the lakeFS Airflow
+commit operator.  We may revisit this decision after feedback from users.
+
+[^1]: Naively the run ID would be sufficient.  But that imposes an
+    assumption of a single Airflow instance.
+
+### Scope
+
+We add to each commit metadata for these items:
+
+* DAG run.  The URL of the DAG run in Airflow that generated this commit.
+  This is the URL of [Get a DAG run][get-dag-run] in the REST API.
+* Optionally, additional DAG identifiers such as:
+  * Git commit digest
+  * Airflow system identifier
+  * Airflow logical ("execution") date
+  * Notes (an Airflow metadata concept)
+  * Conf.
+
+  We allow all items that can be retrieved by [Get a DAG run][get-dag-run]
+  in the REST API, and copy terminology from there.
+
+  All optionals must be exactly that.  Most importantly, conf can contain
+  secrets and _must_ be optional.
```

Comment:
I think if something may contain a secret it should never be logged, not even with a default off option.